### PR TITLE
OCPBUGS-42434: Enable Managed Identity w/ Certificates in HyperShift Control Plane Components

### PR DESCRIPTION
**Pull request by bryan-cox**
**What this PR does / why we need it**:
This PR enables components in the hosted control plane to authenticate with Azure Cloud through client certificate. These components include:
1. KMS
2. CPO
3. CAPZ
4. Cloud Provider

**Which issue(s) this PR fixes**:
Fixes [OCPBUGS-42434](https://issues.redhat.com/browse/OCPBUGS-42434)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: control-plane-operator/controllers/hostedcontrolplane/cloud/azure/reconcile.go [hunk 1]
```diff
@@ -2,6 +2,7 @@ package azure
 
 import (
 	"fmt"
+	"github.com/openshift/hypershift/support/azureutil"
```
Comment (Patryk-Stefanski) [inline control-plane-operator/controllers/hostedcontrolplane/cloud/azure/reconcile.go:5]: Sort imports

File: control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/azure/config.go [hunk 2]
```diff
@@ -49,14 +51,17 @@ func adaptConfigSecret(cpContext component.ControlPlaneContext, secret *corev1.S
 	return nil
 }
 
+func adaptSecretProvider(cpContext component.ControlPlaneContext, secretProvider *secretsstorev1.SecretProviderClass) error {
+	secretProvider = manifests.ManagedAzureSecretProviderClass(secretProvider.Name, cpContext.HCP.Namespace)
```
Comment (muraee) [inline control-plane-operator/controllers/hostedcontrolplane/v2/cloud_controller_manager/azure/config.go:55]:
```suggestion
```
This line is not needed

File: control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go [hunk 3]
```diff
@@ -864,12 +865,26 @@ func healthCheckKASEndpoint(ingressPoint string, port int) error {
...
 		}
 	}
+
+	if hyperazureutil.IsAroHCP() {
+		credentialsSecret := manifests.AzureCredentialInformation(hcp.Namespace)
+		if err := r.Client.Get(ctx, client.ObjectKeyFromObject(credentialsSecret), credentialsSecret); err != nil {
+			return fmt.Errorf("failed to get Azure credentials secret: %w", err)
+		}
+
+		tenantID := strings.TrimSpace(string(credentialsSecret.Data["AZURE_TENANT_ID"]))
```
[the remark this answers is not in the document]
Comment (bryan-cox) [reply control-plane-operator/controllers/hostedcontrolplane/hostedcontrolplane_controller.go:885]: No but I don't think it hurts. I removed it though.

File: hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go [hunk 4]
```diff
@@ -56,18 +59,31 @@ func (a Azure) ReconcileCAPIInfraCR(
...
 
-	if _, err := createOrUpdate(ctx, client, azureCluster, func() error {
+	if _, err := createOrUpdate(ctx, c, azureCluster, func() error {
 		return reconcileAzureCluster(azureCluster, hcluster, apiEndpoint, azureClusterIdentity, controlPlaneNamespace)
 	}); err != nil {
 		return nil, fmt.Errorf("failed to reconcile Azure CAPI cluster: %w", err)
 	}
 
+	// Reconcile the SecretProviderClass
+	nodepoolMgmtSecretProviderClass := manifests.ManagedAzureSecretProviderClass(config.ManagedAzureNodePoolMgmtSecretProviderClassName, controlPlaneNamespace)
```
Comment (enxebre) [inline hypershift-operator/controllers/hostedcluster/internal/platform/azure/azure.go:75]: nit: nodePool

**Pull request merged**